plugin/cypress/support/step_definitions/hooks.ts

```diff
@@ -0,0 +1,127 @@
+import { Before } from '@badeball/cypress-cucumber-preprocessor';
+
+const CLUSTER1_CONTEXT = Cypress.env('CLUSTER1_CONTEXT');
+const CLUSTER2_CONTEXT = Cypress.env('CLUSTER2_CONTEXT');
```

Comment:
Tests don't pass for me locally, bookinfo is not installed, will debug this for more detail.

I think this copypasta might be the issue, I will take a look if we can execute the bookinfo scripts more simpler than utilizing this hook from regular regression - that was maybe a bit over engineered approach 
```
   1) Graph page is displayed correctly
    ✓ Istio Config page is displayed correctly (6999ms)


  3 passing (1m)
  1 failing

  1) Kiali sidebar integration with OCP Console
       Graph page is displayed correctly:
     CypressError: `cy.exec('../_output/kiali/hack/istio/wait-for-namespace.sh -n bookinfo')` failed because the command exited with a non-zero code.

Pass `{failOnNonZeroExit: false}` to ignore exit code failures.

Information about the failure:
Code: 1

Stderr:
Error from server (NotFound): namespaces "kiali-operator" not found

https://on.cypress.io/exec
      at <unknown> (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:119293:77)
      at tryCatcher (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:1807:23)
      at Promise._settlePromiseFromHandler (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:1519:31)
      at Promise._settlePromise (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:1576:18)
      at Promise._settlePromise0 (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:1621:10)
      at Promise._settlePromises (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:1701:18)
      at _drainQueueStep (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:2407:12)
      at _drainQueue (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:2400:9)
      at Async._drainQueues (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:2416:5)
      at Async.drainQueues (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/runner/cypress_runner.js:2286:14)
  From Your Spec Code:
      at Context.eval (https://console-openshift-console.apps.pmarekipi.maistra.upshift.redhat.com/__cypress/tests?p=cypress/e2e/featureFiles/sidebar-navigation.feature:15941:34)

```

Reply:
I have simplified the hook file removing tags that are not used in OSSMC tests and fixed a small issue in OSSMC creation yaml in manage-ossmc.sh script. Anyway these changes will not fix your issue. How have you installed OSSMC? Error says `kiali-operator` namespace is not found, you need the kiali operator installed to install OSSMC right?

Reply:
I do install OSSMC by executing Cypress, which do install OSSMC [via this script ](https://github.com/kiali/openshift-servicemesh-plugin/blob/f8e93a36a41d6d06579057398256e41392de62a5/hack/manage-ossmc.sh#L11). The yaml is copied from [Kiali website](https://kiali.io/docs/installation/installation-guide/creating-updating-ossmconsole-cr/#installing-via-oc-cli) how to install OSSMC, so if I understand this correctly the Kiali operator belongs into `openshift-operators` namespace. 

I tried to modify scripts to execute the cypress hooks in correct namespace, still no luck. Currently I'm provisioning fresh cluster to verify whole downstream pipeline - its possible I have something miss-configured in my environment

Reply:
Yes, you are right. If you install Kiali operator from Openshift OperatorHub the Kiali operator belongs to `openshift-operators` namespace, not the `kiali-operator` one. In that case we will have to create specific `wait-for-namespace` script for OSSMC or modify the current one to include this case.

Reply:
I retested whole pipeline and I again see installed Kiali operator like: 
![image](https://github.com/kiali/openshift-servicemesh-plugin/assets/86114326/3ffbe33d-732f-4261-a4b4-81f4c2da786e)

So there is deployment named `kiali-operator` which do belong into `openshift-operators` namespace

The `wait-for-namespace` and other hacks scripts works for normal regression, so I'm not exactly sure where is the difference, imho we should look into that before modyfing hack scripts?

Reply:
FYI: by default, the Kiali Server will be installed in the same namespace where the Kiali CR is. If you are running in an OLM environment where the operator is installed in "openshift-operators" and you create the CR via the OpenShift Console GUI, the Kiali CR will be created in "openshift-operators". By default, therefore, the Kiali Server will be installed in that namespace, too. If you do not want that behavior, override the default by explicitly setting `spec.deployment.namespace` in the Kiali CR. For the docs on this setting, see: https://kiali.io/docs/configuration/kialis.kiali.io/#.spec.deployment.namespace
```
.spec.deployment.namespace
The namespace into which Kiali is to be installed. If this is empty or not defined, the default will be the namespace where the Kiali CR is located.
```